### ISSUE TITLE
Cannot have nix `0.20` and `0.23` in same package due to conflicting `bitflags` requirement 

### ISSUE THREAD

**Issue by aDotInTheVoid** [nix-rust/nix#1555]
In `Cargo.toml`

```toml
[dependencies]
nix = "0.23.0"
zbus = "1.9.2"
```

This leads to

```rust
error: failed to select a version for `bitflags`.
    ... required by package `nix v0.20.2`
    ... which is depended on by `zbus v1.9.2`
    ... which is depended on by `whyyy v0.1.0 (/private/tmp/whyyy)`
versions that meet the requirements `>=1.1.0, <1.3.0` are: 1.2.1, 1.2.0, 1.1.0

all possible versions conflict with previously selected packages.

  previously selected package `bitflags v1.3.1`
    ... which is depended on by `nix v0.23.0`
    ... which is depended on by `whyyy v0.1.0 (/private/tmp/whyyy)`

failed to select a version for `bitflags` which could resolve this conflict
```

Because `v0.23` needs `bitflags > 1.3` but `v0.20` need `bitflags < 1.3`

**Comment by aDotInTheVoid** [comment]
Also if you could release `0.23.1` with this, that would be super helpfull for fixing the starship issue.

**Comment by asomers** [comment]
Here's my proposed solution:
* vendor the bitflags crate, so there won't _be_ any external dependencies on it.  #1560 
* Set the `rust-version` field in Cargo.toml, which will prevent future problems like this from ever happening again, once the MSRV is 1.56.0 or newer. #1561 
* Create a TODO issue to de-vendor bitflags after the MSRV is eventually raised to 1.56.0.

@aDotInTheVoid what do you think?

**Comment by aDotInTheVoid** [comment]
I think this would solve the problem I'm facing with starship, I'll try in a couple of hours when I'm back at a keyboard

**Comment by asomers** [comment]
Aleksey Kladov came up with a different solution, which we merged as #1593 .
